core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -125,10 +126,46 @@ public CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionD
     }
 
     public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.PartitionData>> acknowledge(
+            String memberId,
             String groupId,
             Map<TopicIdPartition, ShareAcknowledgeRequestData.AcknowledgePartition> acknowledgeTopics
     ) {
-        throw new UnsupportedOperationException("Not implemented yet");
+        log.debug("Acknowledge request for topicIdPartitions: {} with groupId: {}",
+                acknowledgeTopics.keySet(), groupId);
+        Map<TopicIdPartition, CompletableFuture<Errors>> futures = new HashMap<>();
+        acknowledgeTopics.forEach((topicIdPartition, acknowledgePartitionData) -> {
+            SharePartition sharePartition = partitionCacheMap.get(sharePartitionKey(groupId, topicIdPartition));
+            if (sharePartition != null) {
+                synchronized (sharePartition) {
```

Comment:
May be too early to think about share partition synchronization as parallel requests for acknowledge should be thread safe by share partition itself, may be, we can keep this code for now but I don't think we ll need synchronization here.